sphinx/util/inventory.py

```diff
@@ -77,7 +77,9 @@ def read_compressed_lines(self) -> Iterator[str]:
 
 class InventoryFile:
     @classmethod
-    def load(cls: type[InventoryFile], stream: IO, uri: str, joinfunc: Callable) -> Inventory:
+    def load(
+        cls: type[InventoryFile], stream: IO[str], uri: str, joinfunc: Callable,
```

Comment:
```suggestion
        cls: type[InventoryFile], stream: IO[str], uri: str, joinfunc: Callable[[str, str], str],
```

A similar type annotation for the load functions should be done (I cannot comment on an unchanged part of the code).

Reply:
done!